password/src/Data/Password.hs

```diff
@@ -180,24 +195,24 @@ hashPass pass = do
 -- The input 'Salt' and resulting 'PassHash' are both byte-64 encoded.
 --
 -- >>> let salt = Salt "abcdefghijklmnopqrstuvwxyz012345"
--- >>> hashPassWithSalt (Pass "foobar") salt
+-- >>> hashPassWithSalt salt (mkPass "foobar")
 -- PassHash {unPassHash = "14|8|1|YWJjZGVmZ2hpamtsbW5vcHFyc3R1dnd4eXowMTIzNDU=|nENDaqWBmPKapAqQ3//H0iBImweGjoTqn5SvBS8Mc9FPFbzq6w65maYPZaO+SPamVZRXQjARQ8Y+5rhuDhjIhw=="}
 --
 -- (Note that we use an explicit 'Salt' in the example above.  This is so that the
 -- example is reproducible, but in general you should use 'hashPass'.  'hashPass'
 -- generates a new 'Salt' everytime it is called.)
 --
 -- This function uses the hash function from the scrypt package: 'encryptPass'.
-hashPassWithSalt :: Pass -> Salt -> PassHash
-hashPassWithSalt pass salt =
+hashPassWithSalt :: Salt -> Pass -> PassHash
+hashPassWithSalt salt pass =
```

Comment:
I don't mind flipping this, but I think in most instances it shouldn't matter.

As far as I can see, the only time that having `Salt` come first would be helpful is if you wanted to curry `hashPasswithSalt` and hash multiple passwords with the same `Salt`.

However, you shouldn't do this. So I don't think it matters what order these arguments come in for the most part.

That said, I'd be fine keeping it like you have it here.

Reply:
If it's not good to hash different passes with the same salt, would this function have any merit, anyway? Maybe better to just not give the option to make their own `Salt` at all?

Reply:
I like having this function so that there is a pure way to create a `PassHash`, but I agree that maybe there should be a warning added to the documentation about this.